api/src/main/resources/liquibase.xml

```diff
@@ -0,0 +1,320 @@
+			</column>
+			<column name="source" type="int(11)">
+				<constraints nullable="false" unique="true" foreignKeyName="source for idgen_log" references="idgen_identifier_source(id)"/>
+			</column>
+			<column name="identifier" type="varchar(50)">
+				<constraints nullable="false"/>
+			</column>
+			<column name="date_generated" type="datetime">
+				<constraints nullable="false"/>
+			</column>
+			<column name="generated_by" type="int(11)">
+				<constraints nullable="false" unique="true" foreignKeyName="generated_by for idgen_log" references="users(user_id)"/>
+			</column>
+			<column name="comment" type="varchar(255)" defaultValue="NULL">
+				<constraints nullable="true"/>
+			</column>
+		</createTable>
+	</changeSet>
+
+	<changeSet id="17a1d838-bf40-45d0-9e2b-6d177ff7ad62" author="Partners In Health">
+		<preConditions onFail="MARK_RAN">
+			<not><tableExists tableName="idgen_reserved_identifier" /></not>
+		</preConditions>
+		<createTable tableName="idgen_reserved_identifier">
+			<column name="id" type="int(11)" autoIncrement="true">
+				<constraints nullable="false" primaryKey="true" unique="true"/>
+			</column>
+			<column name="source" type="int(11)">
+				<constraints nullable="false" unique="true" foreignKeyName="source for idgen_reserved_identifier" references="idgen_identifier_source(id)"/>
+			</column>
+			<column name="identifier" type="varchar(50)">
+				<constraints nullable="false"/>
+			</column>
+		</createTable>
+	</changeSet>
+
+	<changeSet id="e931b3c1-94b6-4169-90f4-1c6bf49d4883" author="Partners In Health">
+		<preConditions onFail="MARK_RAN">
+			<tableExists tableName="idgen_id_pool" />
+			<not><columnExists tableName="idgen_id_pool" columnName="refill_with_scheduled_task"/></not>
+		</preConditions>
+		<addColumn tableName="idgen_id_pool">
+			<column name="refill_with_scheduled_task" type="tinyint(1)" defaultValue="1">
+				<constraints nullable="false"/>
+			</column>
+		</addColumn>
+	</changeSet>
+
+	<changeSet id="3e32c214-9f9a-4424-8590-43ca68396aa6" author="Partners In Health">
+		<preConditions onFail="MARK_RAN">
+			<tableExists tableName="idgen_remote_source" />
+			<not><columnExists tableName="idgen_remote_source" columnName="user"/></not>
+		</preConditions>
+		<addColumn tableName="idgen_remote_source">
+			<column name="user" type="varchar(50)">
+				<constraints nullable="true"/>
+			</column>
+		</addColumn>
+	</changeSet>
+
+	<changeSet id="1df31052-65ce-4b81-8a9c-86c58c87c042" author="Partners In Health">
+		<preConditions onFail="MARK_RAN">
+			<tableExists tableName="idgen_remote_source" />
+			<not><columnExists tableName="idgen_remote_source" columnName="password"/></not>
+		</preConditions>
+		<addColumn tableName="idgen_remote_source">
+			<column name="password" type="varchar(20)">
+				<constraints nullable="true"/>
+			</column>
+		</addColumn>
+	</changeSet>
+
+	<changeSet id="aea80abc-8e1b-44f2-9574-c64741750cec" author="Jeremy Keiper">
+		<preConditions onFail="MARK_RAN">
```

Comment:
I think in this case we do not need the first precondition because it is implied in the second precondition. Not so?